apps/src/code-studio/pd/workshop_dashboard/components/workshop_enrollment_school_info.jsx

```diff
@@ -47,6 +58,40 @@ export default class WorkshopEnrollmentSchoolInfo extends React.Component {
     this.props.onDelete(pendingDeleteId);
   }
 
+  handleScholarshipStatusChange(enrollment, selection) {
+    this.setState({
+      pendingScholarshipUpdates: [
+        ...this.state.pendingScholarshipUpdates,
+        enrollment.id
+      ]
+    });
+    $.ajax({
+      method: 'POST',
+      url: `/api/v1/pd/enrollment/${enrollment.id}/scholarship_info`,
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify({scholarship_status: selection.value})
+    }).done(data => {
+      this.setState(state => {
+        const enrollments = state.enrollments.map(enrollment => {
+          if (enrollment.id === data.id) {
+            return data;
+          } else {
+            return enrollment;
+          }
+        });
+        return {enrollments};
+      });
+      this.setState(state => {
+        const pendingScholarshipUpdates = state.pendingScholarshipUpdates.filter(
+          e => {
+            return e !== data.id;
+          }
+        );
+        return {pendingScholarshipUpdates};
+      });
+    });
```

Comment:
😮  TIL [functional setState](https://medium.com/@wisecobbler/using-a-function-in-setstate-instead-of-an-object-1f5cfd6e55d1)! Awesome.  We should, in theory, do this in the setState before the ajax call too since it's also building on existing state.  Then, can we just combine these two back-to-back setState calls?